test/persistence-hooks.suite.js

```diff
@@ -1225,6 +1424,52 @@ module.exports = function(dataSource, should) {
           });
       });
 
+      it('triggers `loaded` hook on create', function(done) {
+        TestModel.observe('loaded', pushContextAndNext());
+
+        TestModel.updateOrCreate(
+          { id: 'new-id', name: 'a name' },
+          function(err, instance) {
+            if (err) return done(err);
+
+            if (dataSource.connector.updateOrCreate) {
+              observedContexts.should.eql(aTestModelCtx({
+                data: { id: 'new-id', name: 'a name' }
+              }));
+            } else {
+              observedContexts.should.eql(aTestModelCtx({
+                data: {
+                  id: 'new-id',
+                  name: 'a name'
+                },
+                isNewInstance: true
+              }));
+            }
+            done();
+          });
+      });
+
+      // TODO (pradnya) - Here for Unoptimized connector, 
+      // `pushContextAndNext` is getting called twice. As a result, observedContexts
+      // returns an array and NOT a single instance.
+      // Optimized connector, works as expected.
```

Comment:
I think that's expected because the unoptimized connector calls to operations: "find" and "create", both of them call the "loaded" hook. I am proposing to modify the assertion depending on which connector is being used.

```js
if (connector.updateOrCreate) {
  // assert observedContext is a single item
} else {
  // assert observedContext is an array
}
```

Thoughts?